pkg/virt-controller/services/template.go

```diff
@@ -19,7 +21,27 @@ func (t *templateService) RenderLaunchManifest(vm *v1.VM) (*kubev1.Pod, error) {
 	precond.MustNotBeNil(vm)
 	domain := precond.MustNotBeEmpty(vm.GetObjectMeta().GetName())
 	uid := precond.MustNotBeEmpty(string(vm.GetObjectMeta().GetUID()))
-	True := true
+
+	// VM target container
+	container := kubev1.Container{
+		Name:            "compute",
+		Image:           t.launcherImage,
+		ImagePullPolicy: kubev1.PullIfNotPresent,
+		Command:         []string{"/virt-launcher", "-qemu-timeout", "60s"},
+	}
+
+	// Set up spice ports
+	ports := []kubev1.ContainerPort{}
+	for i, g := range vm.Spec.Domain.Devices.Graphics {
+		if strings.ToLower(g.Type) == "spice" {
+			ports = append(ports, kubev1.ContainerPort{
+				ContainerPort: g.Port,
+				Name:          "spice" + strconv.Itoa(i),
+			})
+		}
+	}
+	container.Ports = ports
```

Comment:
Two things:

1. This prevents us from reusing libvirtd's port allocation. IIUIC libvirtd will tell qemu which port to bind to.
2. We might not need to open the port considering:
```
List of ports to expose from the container. Exposing a port here gives the system additional information about the network connections a container uses, but is primarily informational. Not specifying a port here DOES NOT prevent that port from being exposed. Any port which is listening on the default "0.0.0.0" address inside a container will be accessible from the network. Cannot be updated.
```
from the container spec.